pl_bolts/callbacks/data_monitor.py

```diff
@@ -157,8 +154,7 @@ def __init__(
     def on_train_start(self, trainer: Trainer, pl_module: LightningModule) -> None:
         super().on_train_start(trainer, pl_module)
         submodule_dict = dict(pl_module.named_modules())
-        self._hook_handles = []
```

Comment:
why are we no longer clearing them here?

Reply:
I wouldn't say this was clearing them properly in the first place, the hooks were still installed, we just lost their handles. But yeah, clearing is important in here, @luca-medeiros, can you fix this please?

Reply:
Now that you mentioned about made me rethink my decision for removing it.
I was a bit confused about the `on_train_start`. Thought it would happen only once thus no need for clearance. But it's not always the case. Will fix it!